src/PRNG.h

```diff
@@ -46,43 +54,40 @@ class PRNG {
   // Standard double generation on [0,1.0]
   double operator()() { return (*gen)(); }
 
-  // Generate a double on a [0,b]
-  double rand(double const bound) { return gen->rand(bound); }
+  // Generate a double on [0,bound]
+  double rand(const double bound) { return gen->rand(bound); }
 
-  // Generate a double on a [0,b)
-  double randExc(double const bound) { return gen->randExc(bound); }
+  // Generate a double on [0,bound)
+  double randExc(const double bound) { return gen->randExc(bound); }
 
   // Generate an unsigned int on [0,bound]
   uint randInt(const uint bound) { return (uint)(gen->randInt(bound)); }
 
   // Generate an unsigned int on [0,bound)
   uint randIntExc(const uint bound) { return (uint)(gen->randInt(bound - 1)); }
 
-  // Generates number on (-bound,bound)
-  double Sym(double bound) { return 2 * bound * gen->rand() - bound; }
+  // Generates a double on [-bound,bound]
+  double Sym(const double bound) { return gen->rand(2.0 * bound) - bound; }
+
+  // return between (-bound, bound)
+  double SymExc(double bound) { return gen->randExc(2.0 * bound) - bound; }
 
   /////////////////////////////
   //   GENERATION FUNCTIONS  //
   /////////////////////////////
 
-  XYZ SymXYZ(double bound) {
-    double bound2 = 2 * bound;
-    return XYZ(gen->rand(bound2) - bound, gen->rand(bound2) - bound,
```

Comment:
I changed all the function calls to a MT random number function that are embedded in another function. For instance, here we are using the result of rand() as a parameter to an XYZ object constructor. This was leading to incorrect results -- results where GCC produced a different random number than icc, but moving the calls outside of the constructor and then passing those variables to the constructor led to consistent results. Since this has now happened 3 times under the same circumstances, I decided to change this behavior everywhere it appears in the code.

There are some changes to clarify comments or make them more consistent. And some for formatting purposes, like using 2.0 for a double constant.